storage/innobase/log/log0recv.cc

```diff
@@ -1830,6 +1830,16 @@ static byte *recv_parse_or_apply_log_rec_body(
             !fsp_is_system_or_temp_tablespace(space_id) &&
             /* For cloned db header page has the encryption information. */
             !recv_sys->is_cloned_db) {
+          /* If Encryption info is changed after the last checkpoint lSN, scan
+          the tablespace to know modified pages. This scan and copy of modified
+          pages would avoid the apply the redo logs due to changed in encryption
+          info. As there is problem of apply redo on page 0 because it reset
+          in-memory encryption and page can't decrypted */
+          if (opt_page_tracking && xtrabackup_incremental != nullptr &&
+              recv_sys->recovered_lsn > incremental_start_checkpoint_lsn) {
```

Comment:
shouldn't we be checking here if we have encryption information in memory for this space_id in order to switch to full scan to avoid false positives here?

Reply:
It is not required.  We switch to full scan if we find encryption redo on page 0 which has LSN more than checkpoint LSN, because those page would not track by page-tracking, but these pages are flushed to disk( As alter Encryption forcefully flushed before completing) .

Reply:
what is not clear to me here is how you know that page0 changed from encrypted to non encrypted by just checking incremental_start_checkpoint_lsn and recovered_lsn?

Reply:
in both case, we would switch to a full scan.

Reply:
@altmannmarcelo @rahulmalik87 As of now, mlog_write_string on page 0 is considered as encryption info. Upstream also treats this way. To be more accurate, another way is to determine the offset at which the mlog_write_string happens.